src/api.js

```diff
@@ -272,3 +278,44 @@ api.declare({
     }),
   });
 });
+
+api.declare({
+  name: 'isInstalledFor',
+  title: 'Check if Repository has Integration',
+  description: [
+    'Checks if the integration has been installed for',
+    'a given repository of a given organization or user.',
+  ].join('\n'),
+  stability: 'experimental',
+  method: 'get',
+  route: '/repository/:owner/:repo',
+  output: 'is-installed-for.json',
+}, async function(req, res) {
+  // Extract owner and repo from request into variables
+  let {owner, repo} = req.params;
+
+  // Look up the installation ID in Azure. If no such owner in the table, no error thrown
+  let ownerInfo = await this.OwnersDirectory.load({owner}, true);
+
+  if (ownerInfo) {
+    let instGithub = await this.github.getInstallationGithub(ownerInfo.installationId);
+    let reposList = await instGithub.integrations.getInstallationRepositories();
+
+    // GitHub API returns an array of objects, each of wich has an array of repos
+    // [ { repositories: [ [Object], [Object], [Object] ] } ]
+    console.log('reposList', reposList);
+    try {
+      let installed = reposList.reduce((a, b) => a.repositories.concat(b.repositories), {repositories: []})
```

Comment:
Hmm. That's what I started my morning with, I ran in Node:
```
let d = []
d.reduce((a, b) => a.repositories.concat(b.repositories), {repositories: []})
```
It produced `{ repositories: [] }`